README.md

```diff
@@ -206,19 +206,29 @@ Valet guarantees it will never fail to write to or read from the keychain unless
 * watchOS 2 or later.
 * macOS 10.11 or later.
 
-### Migrating from Valet 2.*
+## Migrating from prior Valet versions
 
-First the good news: you will _not_ have to migrate your keychain data when upgrading from Valet 2.* to Valet 3.*. All Valet objects are backwards compatible with their Valet 2 counterparts. We have exhaustive unit tests to prove it (search for `test_backwardsCompatibility`).
+The good news: most Valet configurations do _not_ have to migrate keychain data when upgrading from and older version of Valet. All Valet objects are backwards compatible with their counterparts from prior versions. We have exhaustive unit tests to prove it (search for `test_backwardsCompatibility`). Valets that have had their configurations deprecated by Apple will need to migrate stored data.
 
-Now the bad news: the Swift Valet API has slight differences from the Objective-C Valet API. You may have noticed a few of the differences in the sample code above, but here’s a rundown of the changes that may affect you.
+The bad news: there are multiple source-breaking API changes from prior versions.
+
+Both guides below explain the changes required to upgrade to Valet 4. 
+
+### Migrating from Valet 2
 
 1. Initializers have changed in both Swift and Objective-C - both languages use class methods now, which felt more semantically honest (a lot of the time you’re not instantiating a new Valet, you’re re-accessing one you’ve already created). [See example usage above](#basic-initialization).
-1. `VALSynchronizableValet` (which allowed keychains to be synced to iCloud) has been replaced by a `Valet.iCloudValet(with:accessibility:)`  (or `+[VALValet iCloudValetWithIdentifier:accessibility:]` in Objective-C). [See examples above](#sharing-secrets-across-devices-with-icloud).
-1. `SecureEnclaveValet` and `SinglePromptSecureEnclaveValet` data retrieval methods now return a single enum [SecureEnclave.Result](Sources/SecureEnclave.swift#L28) rather than using an `inout` boolean to signal whether a user cancelled. The Objective-C API remains the same.
-1. `migrateObjects(matching:)` and `migrateObjects(from:)` now both return a nonnull [MigrationResult](Sources/MigrationResult.swift#L24).
```

Comment:
Presumably there's still a change from how the migrations worked in Valet 2?

Reply:
Correct. But this is covered in the "migrating from Valet 3" guide, and the "migrating from Valet 2" guide tells folks to keep reading.

If that isn't clear, let me know 🙂